store/video.ts

```diff
@@ -0,0 +1,54 @@
+import { useEffect } from "react";
+import { atom, useAtom } from "jotai";
+import { RESET, atomWithReset, useUpdateAtom } from "jotai/utils";
+import type { ResourceSchema } from "$server/models/resource";
+import type { SectionSchema } from "$server/models/book/section";
+import type { VideoInstance } from "$types/videoInstance";
+import { isVideoResource } from "$utils/videoResource";
+import getVideoInstance from "$utils/video/getVideoInstance";
+
+const videoAtom = atomWithReset<{
+  video: Map<ResourceSchema["url"], VideoInstance>;
+  key: ResourceSchema["url"];
+}>({
+  video: new Map(),
+  key: "",
+});
+
+const updateVideoAtom = atom(
+  null,
+  (get, set, sections: Pick<SectionSchema, "topics">[]) => {
+    const { video, key } = get(videoAtom);
+    const resources = sections.flatMap(({ topics }) =>
+      topics.map(({ resource }) => resource)
+    );
+    for (const resource of resources) {
+      if (!isVideoResource(resource)) {
+        video.delete(resource.url);
+        continue;
+      }
+      video.set(resource.url, getVideoInstance(resource));
```

Comment:
変更前の挙動を維持するのであれば自動再生を有効にする必要がある(問題ないか動作確認する)

```suggestion
    const autoplay = true;
    for (const resource of resources) {
      if (!isVideoResource(resource)) {
        video.delete(resource.url);
        continue;
      }
      video.set(resource.url, getVideoInstance(resource, autoplay));
```

Reply:
動作確認して正常に表示時に自動再生されることを確認した。どうして正常なのか理解していない…